core/src/main/java/fr/sncf/osrd/utils/graph/Pathfinding.java

```diff
@@ -74,6 +80,10 @@ public record Range(long start, long end){}
      * Used in STDCM. Either totalDistanceUntilEdgeLocation or edgeRangeCost must be defined.
     */
     private TotalCostUntilEdgeLocation<EdgeT> totalCostUntilEdgeLocation = null;
+    /**
+     * Timeout, in seconds, to avoid infinite loop when no path can be found.
+     */
+    private Long timeout = null;
```

Comment:
If the unit is seconds, we may need decimal values. a float/double seems more appropriate.

If it's changed to float/double, there's no need for an optional value, we can just set the timeout to positive infinity.